### DISCUSSION TITLE
Loading "unaligned" models with proper alignment

### DISCUSSION THREAD

**Discussion by iacore** [rustformers/llama-rs#268] [Ideas]
It's possible to call `mmap(2)` on each tensor separately. This will make every tensors aligned no matter how they are placed in a file.

-- written in case this feature is needed in the future

**Comment by KerfuffleV2** [comment]
I actually had that idea too.

Unfortunately, the offset parameter in `mmap` generally has to be a multiple of the page size. Quoting the man page on Linux: "*Most implementations require that off and addr are multiples of the page size as returned by sysconf().*"

**Reply by iacore** [reply]
this is sad
